frontend/awx/administration/instance-groups/hooks/useInstanceGroupActions.tsx

```diff
@@ -0,0 +1,119 @@
+        ? undefined
+        : t(
+            'You do not have permission to create an instance group. Please contact your organization administrator if there is an issue with your access.'
+          ),
+    [data, t]
+  );
+}
+
+export function useInstanceGroupToolbarActions(
+  onComplete: (instanceGroups: InstanceGroup[]) => void
+) {
+  const { t } = useTranslation();
+  const pageNavigate = usePageNavigate();
+  const deleteInstanceGroups = useDeleteInstanceGroups(onComplete);
+  const isCreateActionDisabled = useDisableCreateInstanceGroup();
+
+  return useMemo<IPageAction<InstanceGroup>[]>(
+    () => [
+      {
+        type: PageActionType.Dropdown,
+        icon: PlusCircleIcon,
+        variant: ButtonVariant.primary,
+        isPinned: true,
+        selection: PageActionSelection.None,
+        label: t('Create group'),
+        isDisabled: isCreateActionDisabled,
+        actions: [
+          {
+            type: PageActionType.Button,
+            selection: PageActionSelection.None,
+            label: t('Create container group'),
+            onClick: () => pageNavigate(AwxRoute.CreateInstanceGroup),
+          },
+          {
+            type: PageActionType.Button,
+            selection: PageActionSelection.None,
+            label: t('Create instance group'),
+            onClick: () => pageNavigate(AwxRoute.CreateInstanceGroup),
+          },
+        ],
+      },
+      { type: PageActionType.Seperator },
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Multiple,
+        icon: TrashIcon,
+        label: t('Delete selected instance groups'),
+        onClick: deleteInstanceGroups,
+        isDanger: true,
+      },
+    ],
+    [deleteInstanceGroups, isCreateActionDisabled, pageNavigate, t]
+  );
+}
+
+export function useInstanceGroupRowActions(onComplete: (instanceGroups: InstanceGroup[]) => void) {
+  const { t } = useTranslation();
+  const pageNavigate = usePageNavigate();
+  const deleteInstanceGroups = useDeleteInstanceGroups(onComplete);
+
+  return useMemo<IPageAction<InstanceGroup>[]>(
+    () => [
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Single,
+        icon: EditIcon,
+        isPinned: true,
+        label: t('Edit instance group'),
+        isDisabled: (instanceGroup) =>
+          instanceGroup.summary_fields?.user_capabilities?.edit
+            ? undefined
+            : t(
+                'You do not have permission to edit this instance group. Please contact your organization administrator if there is an issue with your access.'
+              ),
+        onClick: (instanceGroup) =>
+          pageNavigate(AwxRoute.EditInstanceGroup, { params: { id: instanceGroup.id } }),
+      },
+      { type: PageActionType.Seperator },
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Single,
+        icon: TrashIcon,
+        label: t('Delete instance group'),
```

Comment:
On these row actions we should distinguish between instance groups and container groups.